kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/VectorSpace.kt

```diff
@@ -0,0 +1,47 @@
+package scientifik.kmath.geometry
+
+import scientifik.kmath.dimensions.Dimension
+import scientifik.kmath.linear.Point
+import scientifik.kmath.operations.ExtendedField
+import scientifik.kmath.operations.Space
+
+interface Vector<T, D : Dimension> : Point<T>
+
+interface VectorSpace<T, D : Dimension, V : Vector<T, D>> : Space<V> {
+    fun vectorFrom(x: Sequence<T>): V
+
+    // TODO what about it?
+    fun multiply(a: V, k: T): V = multiply(a, k as Number)
```

Comment:
Unchecked cast to Number looks weird. Isn't it possible to add type constraint in the interface decl.:
`interface VectorSpace<T : Number, D : Dimension, V : Vector<T, D>> : Space<V> `
?

Reply:
probably the best option would be not to provide a default implementation, and child classes would be able to implement it with help of `RingOperations::multiply(T, T):T`

Reply:
Yes, I think it is the best option

Reply:
done